CMSgov/design-system
# [WNMGDS-2811] Fix body scroll freeze persisting after dialog unmounts

## Summary

WNMGDS-2811

In cases where a dialog unmounts without being able to set its `isOpen` prop to false, our dialog element wouldn't clean up its scroll-preventing additions to the body element. This could happen in single-page applications using React routing where and the user navigates away from a page with an open dialog. Unfortunately there's no clean way I can access state in a `useEffect` cleanup function that doesn't have any dependencies (when we know it's unmounting for no other reason than the parent is unmounting), so I simplified everything to remove the need for referencing state when "unfreezing".

## How to test

1. Check out 30ada9413a32f218eabd918665c0ee37d602255b
2. Open http://localhost:6006/iframe.html?id=components-dialog--prevent-scroll-example&viewMode=story
3. Scroll down and click the button to open the dialog, and then click the blue button to de-render the parent. In this commit with the fix, you should be able to scroll the page normally. If you were to check out b0d0a59e033cb738fd48d9d6b76498490c48a9be without the fix, the page would be frozen.

## Checklist


### If this is a change to code:
